cmd/state-installer/internal/installer/installation_darwin.go

```diff
@@ -2,7 +2,58 @@
 
 package installer
 
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/appinfo"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/gobuffalo/packr"
+)
+
+const appDir = "/Applications"
+
 // InstallSystemFiles installs files in the /Application directory
 func InstallSystemFiles(installDir string) error {
+	box := packr.NewBox("../../../../assets/macOS/app")
+	err := box.Walk(func(path string, _ packr.File) error {
+		if fileutils.IsDir(path) {
+			err := fileutils.Mkdir(filepath.Join(appDir, path))
```

Comment:
I think in an update, this would error out, because the directory already exists.  We somehow need to make this function able to run twice without error. The easiest way is probably to just remove the entire target directory "ActiveState Desktop.app" at the beginning.

Of course it would be nice to also have a rollback function for this, but that is probably better done in a follow-up story, as my base PRs haven't landed yet.  And it will be easier once that has happened.

Reply:
Actually, this function will just return `nil` if the directory already exists. But I think it's better to just remove the directory and all of its contents rather than overwriting everything so I've applied this suggestion.

Is there a follow-up story for rollback functionality in our update/install mechanism already? If not I can create one.

Reply:
Ticket is here: https://www.pivotaltracker.com/story/show/177873388